libraries/aws/shadow/integration-test/shadow_system_test.c

```diff
@@ -0,0 +1,818 @@
+
+    /* The interval at which an MQTT PINGREQ needs to be sent out to broker. */
+    connectInfo.keepAliveSeconds = MQTT_KEEP_ALIVE_INTERVAL_SECONDS;
+
+    /* Username and password for authentication. Not used in this test. */
+    connectInfo.pUserName = NULL;
+    connectInfo.userNameLength = 0U;
+    connectInfo.pPassword = NULL;
+    connectInfo.passwordLength = 0U;
+
+    /* Send MQTT CONNECT packet to broker. */
+    TEST_ASSERT_EQUAL( MQTTSuccess, MQTT_Connect( pContext,
+                                                  &connectInfo,
+                                                  NULL,
+                                                  CONNACK_RECV_TIMEOUT_MS,
+                                                  pSessionPresent ) );
+}
+
+static void eventCallback( MQTTContext_t * pContext,
+                           MQTTPacketInfo_t * pPacketInfo,
+                           MQTTDeserializedInfo_t * pDeserializedInfo )
+{
+    ShadowMessageType_t messageType = ShadowMessageTypeMaxNum;
+    const char * pThingName = NULL;
+    uint16_t thingNameLength = 0U;
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+    uint16_t packetIdentifier;
+
+    assert( pDeserializedInfo != NULL );
+    assert( pContext != NULL );
+    assert( pPacketInfo != NULL );
+
+    packetIdentifier = pDeserializedInfo->packetIdentifier;
+
+    /* Handle incoming publish. The lower 4 bits of the publish packet
+     * type is used for the dup, QoS, and retain flags. Hence masking
+     * out the lower bits to check if the packet is publish. */
+    if( ( pPacketInfo->type & 0xF0U ) == MQTT_PACKET_TYPE_PUBLISH )
+    {
+        assert( pDeserializedInfo->pPublishInfo != NULL );
+        /* Handle incoming publish. */
+
+        /* Handle incoming publish. */
+        /* Let the Device Shadow library tell us whether this is a device shadow message. */
+        if( SHADOW_SUCCESS == Shadow_MatchTopic( pDeserializedInfo->pPublishInfo->pTopicName,
+                                                 pDeserializedInfo->pPublishInfo->topicNameLength,
+                                                 & messageType,
+                                                 & pThingName,
+                                                 & thingNameLength ) )
+        {
+            /* Upon successful return, the messageType has been filled in. */
+            if( messageType == ShadowMessageTypeUpdateDelta ) 
+            {
+                receivedUpdateDeltaResult = true;
+                LogInfo(( "/update/delta payload:%s.\n\n", ( const char * ) pDeserializedInfo->pPublishInfo->pPayload ));
+                /* validate the each field in json payload if meet our expectation. */
+            } 
+            else if ( ( messageType == ShadowMessageTypeUpdateDocuments ) )
+            {
+                receivedUpdateDocumentsResult = true;
+                LogInfo(( "/update/documents json payload:%s.\n\n", ( const char * ) pDeserializedInfo->pPublishInfo->pPayload ));
```

Comment:
If you run uncrustify there will be spaces added here

Reply:
I could run uncrustify, this will be fixed at next commit.